pattern/constraint/thing/RelationConstraint.java

```diff
@@ -149,8 +156,8 @@ public AlphaEquivalence alphaEquals(RelationConstraint that) {
         private final int hash;
 
         public RolePlayer(@Nullable TypeVariable roleType, ThingVariable player, int repetition) {
-            assert roleType == null || roleType.reference().isName() ||
-                    (roleType.label().isPresent() && roleType.label().get().scope().isPresent());
+//            assert roleType == null || roleType.reference().isName() ||
+//                    (roleType.label().isPresent() && roleType.label().get().scope().isPresent()); // TODO We have a problem here if we reach this before we've added a labelconstraint to the role type variable
```

Comment:
umm hmmmm hmmmm hmmm So I think it would actually be quite dangerous to remove this check, because we can rely on it to make sure we've made a _valid_ clone. However this pushes us towards option 2... in terms of tradeoff having something safer vs something somewhat ugly I'd rather take the safer option? It may well end up that we have to define a total order of the constraint cloning (eg. labels first, then value, the has, etc. etc.)